File: leetcode/Python/guess-number-higher-or-lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        r, l, m, t = n, 1, 0, 0
        while l<=r:
            m = (l+r)//2
            t = guess(m)
            if t < 0:
                r = m-1
            elif t > 0:
                l = m+1
            else:
                break
        return m

File: leetcode/C++/guess-number-higher-or-lower.cpp
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * int guess(int num);
 */

class Solution {
public:
    int guessNumber(int n) {
        long r=n, l=1, m=0, t=0;
        while(l<=r){
            m=(l+r)/2;
            if((t=guess(m))<0) r=m-1;
            else if(t>0) l=m+1;
            else break;
        }
        return m;
    }
};